src/components/VddkUploader/utils/utils.ts

```diff
@@ -0,0 +1,56 @@
+import { getInventoryApiUrl } from 'src/modules/Providers/utils/helpers/getApiUrl';
+
+import { consoleFetch } from '@openshift-console/dynamic-plugin-sdk';
+import { PHASES } from '@utils/constants';
+import { t } from '@utils/i18n';
+
+import {
+  type UploadTarballResponse,
+  type VddkBuild,
+  type VddkBuildResponse,
+  vddkBuildResponseVariant,
+} from './types';
+
+export const uploadVddkTarball = async (file: File): Promise<UploadTarballResponse> => {
+  const formData = new FormData();
+  formData.append('file', file, file.name);
+
+  const response = await consoleFetch(getInventoryApiUrl('vddk/build-image'), {
+    body: formData,
+    method: 'POST',
+  });
+
+  return response.json() as Promise<UploadTarballResponse>;
+};
+
+export const getVddkImageBuildResponse = (status: VddkBuild['status']): VddkBuildResponse => {
+  if (status?.phase === PHASES.COMPLETE) {
+    const body = status?.outputDockerImageReference.replace(
+      ':latest',
+      `@${status?.output?.to?.imageDigest}`,
+    );
+    return {
+      body,
+      variant: vddkBuildResponseVariant.success,
+    };
+  }
+
+  if (status?.phase === PHASES.FAILED) {
+    return {
+      body: status?.logSnippet ?? '',
+      title: t('Failed'),
+      variant: vddkBuildResponseVariant.warning,
+    };
+  }
+
+  return {
+    body: t('Building image'),
+    variant: vddkBuildResponseVariant.info,
+  };
+};
+
+export const getUploadButtonText = (uploading: boolean, buildingImage: boolean) => {
```

Comment:
this can be a switch case / object , then u need to send only 1 arg

Reply:
I don't think I follow, the 2 args are not really connected
uploading is a flag to know if the file is currently uploading or not.
and buildingImage is a flag to follow if the image is still building.